cli/src/common/aws/aws_resources.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+from common.aws.aws_api import AWSApi
+from common.boto3.common import AWSClientError
+from pcluster.utils import get_region
+
+
+class StackActionError(Exception):
+    """Represent an error during the execution of an action on the stack."""
+
+    def __init__(self, message: str):
+        super().__init__(message)
+
+
+class Stack:
+    """Object to interact with a Stack, initialized with a describe_stacks call."""
+
+    def __init__(self, stack_name: str):
+        """
+        Init StackInfo by performing a describe_stacks call.
+
+        If the stack doesn't exist it raises an exception.
+        """
+        self.name = stack_name
+        self.region = get_region()
+        try:
+            self._stack_data = AWSApi().instance().cfn.describe_stack(stack_name)
+        except AWSClientError as e:
+            if "does not exist" in str(e):
+                raise StackActionError(f"The stack {stack_name} does not exist.")
```

Comment:
Why creating another Exception class, instead of using AWSClientError?

Reply:
Because I want to manage StackActionError differently from AWSClientError.
When getting a StackActionError "the **stack** pcluster-yyy doesn't exist" I can catch it and show: "the **cluster** yyy doesn't exist".